### PR TITLE
Keep the resource and property names in a compressed binary file

### PR DESCRIPTION
**Pull request by henriquegemignani**
zstd level 15 was one of the fastest decompression algorithms, with the best compression ratio.

### CODE REVIEW

File: src/mercury_engine_data_structures/_dread_data_construct.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            construct.Int32un,
+            construct.Sequence(
+                construct.PascalString(construct.Int16un, "ascii"),  # key
+                construct.Int64un,  # hash
+            )
+        )
+
+    def _parse(self, stream, context, path) -> typing.Dict[str, int]:
+        key_size_struct = struct.Struct("=H")
+        value_size_struct = struct.Struct("=Q")
```
Comment (duncathan) [inline src/mercury_engine_data_structures/_dread_data_construct.py:34]: this isn't a size, it's just the value

File: tools/create_class_definitions.py [hunk 2]
```diff
@@ -224,6 +232,47 @@ def export_code(self):
...
+        super().__init__(subcon)
+        import zstd
+        self.lib = zstd
+        self.level = level
+
+    def _decode(self, data, context, path):
+        return self.lib.decompress(data)
+
+    def _encode(self, data, context, path):
+        return self.lib.compress(data, self.level)
```
Comment (duncathan) [inline tools/create_class_definitions.py:246]: is this not duplicated code?

**Review by duncathan — CHANGES_REQUESTED** [review]

File: tools/create_class_definitions.py (hunk 2 above)
Comment (henriquegemignani) [reply tools/create_class_definitions.py:246]: oh, yeah could move to construct_extensions

Comment (henriquegemignani) [reply tools/create_class_definitions.py]: Better yet, unused duplicated code :)

**Review by duncathan — APPROVED** [review]

**Pull request merged**